[2025-07-25T19:08:47.123+0700] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-07-25T19:08:47.873+0700] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: etl_local_postgres_pipeline.load_songplays_fact_table manual__2025-07-25T11:44:30.085491+00:00 [queued]>
[2025-07-25T19:08:48.080+0700] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: etl_local_postgres_pipeline.load_songplays_fact_table manual__2025-07-25T11:44:30.085491+00:00 [queued]>
[2025-07-25T19:08:48.081+0700] {taskinstance.py:2306} INFO - Starting attempt 2 of 4
[2025-07-25T19:08:48.358+0700] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): load_songplays_fact_table> on 2025-07-25 11:44:30.085491+00:00
[2025-07-25T19:08:48.378+0700] {standard_task_runner.py:90} INFO - Running: ['airflow', 'tasks', 'run', 'etl_local_postgres_pipeline', 'load_songplays_fact_table', 'manual__2025-07-25T11:44:30.085491+00:00', '--job-id', '84', '--raw', '--subdir', 'DAGS_FOLDER/etl_pipeline_pjt.py', '--cfg-path', '/tmp/tmp4jninjzz']
[2025-07-25T19:08:48.380+0700] {standard_task_runner.py:91} INFO - Job 84: Subtask load_songplays_fact_table
[2025-07-25T19:08:48.380+0700] {logging_mixin.py:188} WARNING - /home/ngtph/airflow/airflow_venv/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=20067) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-07-25T19:08:48.380+0700] {standard_task_runner.py:63} INFO - Started process 20071 to run task
[2025-07-25T19:08:51.882+0700] {task_command.py:426} INFO - Running <TaskInstance: etl_local_postgres_pipeline.load_songplays_fact_table manual__2025-07-25T11:44:30.085491+00:00 [running]> on host DESKTOP-9M5T9P5.localdomain
[2025-07-25T19:08:55.523+0700] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='phuongnt226' AIRFLOW_CTX_DAG_ID='etl_local_postgres_pipeline' AIRFLOW_CTX_TASK_ID='load_songplays_fact_table' AIRFLOW_CTX_EXECUTION_DATE='2025-07-25T11:44:30.085491+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-07-25T11:44:30.085491+00:00'
[2025-07-25T19:08:55.526+0700] {taskinstance.py:430} INFO - ::endgroup::
[2025-07-25T19:08:55.542+0700] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-07-25T19:08:55.542+0700] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/ngtph/airflow/airflow_venv/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ngtph/airflow/airflow_venv/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ngtph/airflow/airflow_venv/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 400, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ngtph/airflow/airflow_venv/lib/python3.12/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ngtph/airflow/airflow_venv/lib/python3.12/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ngtph/airflow/dags/etl_pipeline_pjt.py", line 86, in load_songplays_table
    execute_sql(sql_queries.songplay_table_insert)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: module 'sql_queries' has no attribute 'songplay_table_insert'
[2025-07-25T19:08:56.255+0700] {taskinstance.py:1206} INFO - Marking task as UP_FOR_RETRY. dag_id=etl_local_postgres_pipeline, task_id=load_songplays_fact_table, run_id=manual__2025-07-25T11:44:30.085491+00:00, execution_date=20250725T114430, start_date=20250725T120847, end_date=20250725T120856
[2025-07-25T19:08:56.561+0700] {standard_task_runner.py:110} ERROR - Failed to execute job 84 for task load_songplays_fact_table (module 'sql_queries' has no attribute 'songplay_table_insert'; 20071)
[2025-07-25T19:08:56.604+0700] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2025-07-25T19:08:57.693+0700] {local_task_job_runner.py:222} INFO - ::endgroup::
